### PR TITLE
feat: bump TypeScript to 4.0.5

### PR DESCRIPTION
**Pull request by trivikr**
TypeScript 4.0.5 release notes https://github.com/microsoft/TypeScript/releases/tag/v4.0.5
TypeScript upgrade steps followed from https://github.com/denoland/deno/blob/master/cli/dts/README.md

### CODE REVIEW

File: cli/dts/lib.esnext.d.ts [hunk 1]
```diff
@@ -22,4 +22,3 @@ and limitations under the License.
 /// <reference lib="esnext.intl" />
 /// <reference lib="esnext.string" />
 /// <reference lib="esnext.promise" />
-/// <reference lib="esnext.weakref" />
```
Comment (kitsonk) [inline cli/dts/lib.esnext.d.ts:25]: This was a manual patch until it is available in TypeScript (see: #8056)... Please do not remove.

**Review by kitsonk — CHANGES_REQUESTED** [review]

File: cli/dts/lib.esnext.d.ts (hunk 1 above)
Comment (trivikr) [reply cli/dts/lib.esnext.d.ts]: reintrouduced in [`a9d6dae` (#8138)](https://github.com/denoland/deno/pull/8138/commits/a9d6dae2edc2c99d63a50ac9244f18b83e03e903)

**Review by crypt096 — APPROVED** [review]

**Comment by bartlomieju** [comment]
@kitsonk please review and land

**Review by kitsonk — APPROVED** [review]
LGTM

**Pull request merged**